README.md

```diff
@@ -0,0 +1,60 @@
+# PSR-7 Storage-less HTTP Sessions - Zend Expressive Session Integration
+
+[![Build Status](https://travis-ci.org/psr7-sessions/storageless-zend-expressive-integration.svg)](https://travis-ci.org/psr7-sessions/storageless-zend-expressive-integration)
+[![Scrutinizer Code Quality](https://scrutinizer-ci.com/g/psr7-sessions/storageless-zend-expressive-integration/badges/quality-score.png?b=master)](https://scrutinizer-ci.com/g/psr7-sessions/storageless-zend-expressive-integration/?branch=master)
+[![Code Coverage](https://scrutinizer-ci.com/g/psr7-sessions/storageless-zend-expressive-integration/badges/coverage.png?b=master)](https://scrutinizer-ci.com/g/psr7-sessions/storageless-zend-expressive-integration/?branch=master)
+[![Packagist](https://img.shields.io/packagist/v/psr7-sessions/storageless-zend-expressive-integration.svg)](https://packagist.org/packages/psr7-sessions/storageless-zend-expressive-integration)
+[![Packagist](https://img.shields.io/packagist/vpre/psr7-sessions/storageless-zend-expressive-integration.svg)](https://packagist.org/packages/psr7-sessions/storageless-zend-expressive-integration)
+
+This integration allows you to use storageless as an implementation for [zend-expressive-session][1]
+
+#### Installation
+
+```sh
+composer require lcobucci/clock \
+                 psr7-sessions/storageless \
+                 psr7-sessions/storageless-zend-expressive-integration
+```
+
+#### Symmetric key
+
+```php
+use Lcobucci\Clock\SystemClock;
+use PSR7Sessions\Storageless\Http\SessionMiddleware as PSR7SessionMiddleware;
+use PSR7Sessions\ZendExpressive\Storageless\SessionPersistence;
+use Zend\Expressive\Session\SessionMiddleware;
+
+$app = \Zend\Expressive\AppFactory::create();
+$app->pipe(PSR7SessionMiddleware::fromSymmetricKeyDefaults(
+    'OpcMuKmoxkhzW0Y1iESpjWwL/D3UBdDauJOe742BJ5Q=',
+    1200
+));
+$app->pipe(new SessionMiddleware(new SessionPersistence(new SystemClock())));
+```
+
+#### Asymmetric key
+
+```php
+use Lcobucci\Clock\SystemClock;
+use PSR7Sessions\Storageless\Http\SessionMiddleware as PSR7SessionMiddleware;
+use PSR7Sessions\ZendExpressive\Storageless\SessionPersistence;
+use Zend\Expressive\Session\SessionMiddleware;
+
+$app = \Zend\Expressive\AppFactory::create();
+$app->pipe(PSR7SessionMiddleware::fromSymmetricKeyDefaults(
+    file_get_contents('/path/to/private_key.pem'),
+    file_get_contents('/path/to/public_key.pem'),
+    1200
+));
+$app->pipe(new SessionMiddleware(new SessionPersistence(new SystemClock())));
+```
+
+[1]: https://github.com/zendframework/zend-expressive-session
```

Comment:
We'll probably need to move all these to use mezzio

Reply:
i suggest to finish this PR and use the migration tool as far as possible before the first release

Reply:
@Ocramius let us finish this PR, and then i'll take care about the laminas migration